src/components/menu/Menu.js

```diff
@@ -47,15 +47,15 @@ class Menu extends React.Component<Props> {
         this.setInitialFocusIndex();
     }
 
-    UNSAFE_componentWillReceiveProps(nextProps: Props) {
-        if (nextProps.isSubmenu && !nextProps.isHidden && this.props.isHidden) {
-            // If updating submenu, use the upcoming props instead of current props.
+    componentDidUpdate(prevProps: Props) {
+        if (this.props.isSubmenu && prevProps.isHidden && !this.props.isHidden) {
+            // If updating submenu, use the current props instead of previous props.
             this.setMenuItemEls();
-            this.setInitialFocusIndex(nextProps);
+            this.setInitialFocusIndex(this.props);
+
+            return;
```

Comment:
Can we avoid this naked return by reorganizing the logic? It looks like `this.setMenuItemEls()` is also called below.